### PR TITLE
Update to qemu 5.1

### PR DESCRIPTION
**Pull request by kernle32dll**

### CODE REVIEW

**Comment by hardillb** [comment]
The resize to 2GB is only going to work with the Lite image by the look of things

**Comment by hardillb** [comment]
And it will need to check the size of a passed in image as well for the VM only build.

File: Dockerfile [hunk 1]
```diff
@@ -89,8 +90,8 @@ ENTRYPOINT ["./entrypoint.sh"]
 # It's just the VM image with a compressed Raspbian filesystem added
 FROM dockerpi-vm as dockerpi
 LABEL maintainer="Luke Childs <lukechilds123@gmail.com>"
-ARG FILESYSTEM_IMAGE_URL="http://downloads.raspberrypi.org/raspbian_lite/images/raspbian_lite-2019-09-30/2019-09-26-raspbian-buster-lite.zip"
-ARG FILESYSTEM_IMAGE_CHECKSUM="a50237c2f718bd8d806b96df5b9d2174ce8b789eda1f03434ed2213bbca6c6ff"
+ARG FILESYSTEM_IMAGE_URL="http://downloads.raspberrypi.org/raspbian_lite/images/raspbian_lite-2020-02-14/2020-02-13-raspbian-buster-lite.zip"
```
Comment (lukechilds) [inline Dockerfile:93]: I'm assuming this change isn't required to update to QEMU 5.1? If so, best to update the image in a separate PR.

**Review by lukechilds — CHANGES_REQUESTED** [review]

File: entrypoint.sh [hunk 2]
```diff
@@ -10,6 +10,8 @@ if [ ! -e $image_path ]; then
       echo "Extracting fresh filesystem..."
       unzip $zip_path
       mv -- *.img $image_path
+
+      qemu-img resize $image_path 2G
```
Comment (lukechilds) [inline entrypoint.sh:14]: What is the reasoning behind this? Just to increase the filesystem size for normal use? Or is it required for QEMU 5.1?

If not required, it should be a separate PR.

Comment (hardillb) [reply entrypoint.sh:14]: This is required for 64 bit based CPUs and is required for all at qemu 5.2

**Comment by hardillb** [comment]
The other PR #21  is better suited to be merged as it adds all this and network support the Pi2 and Pi3 based systems and jumps straight to qemu 5.2

File: entrypoint.sh (hunk 2 above)
Comment (lukechilds) [reply entrypoint.sh:14]: Thanks!

So there's just a hard limit that all images must be no less that 2GB in size? Do you have source you could point me to so I can read more about that?

**Comment by lukechilds** [comment]
Yeah I've seen that PR, it looks great!

However I prefer to keep PRs down to a single atomic change. So one change = one PR = one squashed commit merged into master.

I've seen QEMU 6.0.0 is out so just testing that locally. Once that's merged it would be great to rebase #21 on to master just adding network support for the Pis without changing QEMU or Raspbian images.

File: entrypoint.sh (hunk 2 above)
Comment (hardillb) [reply entrypoint.sh:14]: It's been so long I can't remember if it's in the docs but I know it fails with an error message if you try with anything that is not a multiple of 2g

**Comment by hardillb** [comment]
But 5.2 is needed for the network change, so nearly everything in the other PR is required

File: Dockerfile (hunk 1 above)
Comment (hardillb) [reply Dockerfile:93]: No, but given how old the previous image was it made sense to move up as well.

File: entrypoint.sh (hunk 2 above)
Comment (lukechilds) [reply entrypoint.sh:14]: Found it: mail-archive.com/qemu-devel@nongnu.org/msg723028.html

**Comment by lukechilds** [comment]
Fair point, ok lets merge it as one PR then.

**Comment by lukechilds** [comment]
Closed in favour of https://github.com/lukechilds/dockerpi/pull/21
